signal_interpolation_fourier.py

```diff
@@ -0,0 +1,589 @@
+        if self.method == "timing":
+            if verbose: print('Doing freq dependent timing...')
+            # Get arrival times in a sliding frequency window
+            self.freq_dependent_timing = self.get_freq_dependent_timing_correction(upsample_factor=upsample_factor, ignore_cutoff_freq_in_timing=ignore_cutoff_freq_in_timing)
+            if verbose: print('Done freq dependent timing')
+        else:
+            # print('NOT doing freq dependent timing')
+            self.freq_dependent_timing = np.zeros(self.phasespectrum_corrected.shape)
+
+        # Remove sliding-window timings from phase spectra
+        for i, ant in enumerate(range(Nants)):
+            for pol in range(Npols):
+                this_phase_corrections = 2*np.pi * (self.freqs*1.0e6) * self.freq_dependent_timing[ant, :, pol]
+                self.phasespectrum_corrected[ant, :, pol] = self.phasespectrum_corrected[ant, :, pol] + this_phase_corrections # this can be done more efficiently...
+
+        self.coherency = self.degree_of_coherency()
+
+        """
+        Produce interpolators for amplitude and phase spectrum
+        Do the Fourier interpolation for each frequency bin < 500 MHz, and for each polarization, separately
+        Note: an order of magnitude speed improvement should be obtainable by vectorizing the Fourier interpolator
+        """
+        nof_freq_channels = len(np.where( (self.freqs < 500) )[0]) # remove hardcoded number
+
+        self.interpolators_abs_spectrum = np.empty( (Npols, nof_freq_channels), dtype=object)# [ [None]*nof_freq_channels ] * Npols
+        #self.interpolators_phase = np.empty( (Npols, nof_freq_channels), dtype=object)#
+
+        """
+        Create interpolators for the "phasors" for each frequency,
+        i.e. exp(i phi(f)) = (cos(i phi), sin(i phi)) for each frequency
+        """
+        self.interpolators_cosphi = np.empty( (Npols, nof_freq_channels), dtype=object)
+        self.interpolators_sinphi = np.empty( (Npols, nof_freq_channels), dtype=object)
+
+        self.interpolators_freq_dependent_timing = np.empty( (Npols, nof_freq_channels), dtype=object)
+
+        self.interpolators_timing = np.empty(Npols, dtype=object)
+
+        self.interpolators_constphase = np.empty(Npols, dtype=object)
+
+        self.interpolators_cutoff_freq = np.empty(Npols, dtype=object)
+
+        if verbose: print('Creating %d interpolators total' % (3*Npols*nof_freq_channels + 3*Npols), end=' ')
+
+        # Create and initialize the interpolators for all quantities
+        for freq_channel in range(nof_freq_channels):
+            for pol in range(Npols):
+                self.interpolators_abs_spectrum[pol, freq_channel] = interpF.interp2d_fourier(x, y, self.abs_spectrum[:, freq_channel, pol])
+                #self.interpolators_phase[pol, freq_channel] = interpF.interp2d_fourier(x, y, self.phasespectrum_corrected[:, freq_channel, pol])
```

Comment:
See above. People may wonder with this code is here, if it is not used.

Reply:
removed